lib/Yield/Assembly/DisassemblyHelper.cpp

```diff
@@ -0,0 +1,154 @@
+/// \file DisassemblyHelper.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "revng/EarlyFunctionAnalysis/FunctionMetadata.h"
+#include "revng/Model/Binary.h"
+#include "revng/Model/Function.h"
+#include "revng/Model/RawBinaryView.h"
+#include "revng/Support/Debug.h"
+#include "revng/Yield/Assembly/DisassemblerInstance.h"
+#include "revng/Yield/Assembly/DisassemblyHelper.h"
+
+namespace detail {
+
+class HiddenDisassemberStorage
+  : public std::map<MetaAddressType::Values, DisassemblerInstance> {};
```

Comment:
You sure you want inheritance as opposed to composition?

Reply:
Yep, the storage IS the map.

The whole point of even having a separate name is to hide the template parameters of the map from the header.